src/main/java/Duke.java

```diff
@@ -94,13 +94,13 @@ public static void main(String[] args) {
 
             } else if (inputMessage.contains("deadline")) {
                 String deadlineName = parsedMessages[1].substring(0, parsedMessages[1].indexOf("/") - 1);
-                String deadlineBy = parsedMessages[1].substring(parsedMessages[1].lastIndexOf(" ") + 1);
+                String deadlineBy = parsedMessages[1].substring(parsedMessages[1].indexOf("/") + 4);
                 addTask(new Deadline(deadlineName, deadlineBy));
                 userTasks[taskCount - 1].printAcknowledgeMessage();
 
             } else if (inputMessage.contains("event")) {
                 String eventName = parsedMessages[1].substring(0, parsedMessages[1].indexOf("/") - 1);
-                String eventTime = parsedMessages[1].substring(parsedMessages[1].lastIndexOf(" ") + 1);
+                String eventTime = parsedMessages[1].substring(parsedMessages[1].indexOf("/") + 4);
```

Comment:
Maybe you can refactor this into a separate method, so you can reuse the code for both event and deadline methods. This may improve the code quality